### PR TITLE
feat: resize on expand collapse

### PR DESCRIPTION
**Pull request by niekvanstaveren**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -121,19 +124,22 @@ export default function supernova(env) {
...
+              styling,
+              setExpandedCallback,
+              wrapperState,
+              selectionObj,
+              useTransitions: expandedState.useTransitions,
+              element,
+            });
+          }
+        },
+        layout.resizeOnExpand ? [containerData, selectionObj.state] : [expandedState, containerData, selectionObj.state]
```
Comment (Caele) [inline src/index.js]: hmm, not sure you are allowed to do conditionals like this for hooks. Can't you just do this check in the "if" surrounding the paintTree call instead?

Comment (niekvanstaveren) [reply src/index.js]: It seems to work, but I'll convert since it's not very clear to read in this way

Comment (Caele) [reply src/index.js]: I think it depends on the internal implementation of hooks, for example if the dependencies are updated on each useEffect call or not. You could achieve the same thing with a flagged state, that only updates if resizeOnExpand is true, but still, clearer if you just check it inside the hook. Performance wise it should just be a single extra function call each loop.

Comment (niekvanstaveren) [reply src/index.js]: Moved expanded state into its own useEffect cycle

**Pull request merged**